### PR TITLE
runtime: test that signext and saturating ops are disabled

### PR DESCRIPTION
**Pull request by nagisa**
These are not currently supported and any decision to enable them should
be a intentional decision, not an accidental one.

Overall these changes are no-op since the deserialization will fail
earlier when trying to deserialize with pwasm deserializer, but if and
when we eventually get rid of it, it would be nice if these extensions
didn't accidentally get enabled.

This puts to rest the question raised in https://github.com/near/nearcore/pull/8886#discussion_r1159879325.

### CODE REVIEW

**Review by jakmeier — COMMENTED** [review]
Thanks for coming back to this! I agree it's better to have them disabled if they weren't supported and if we would fail parsing them earlier anyway.

File: runtime/near-vm-runner/src/tests/compile_errors.rs [hunk 1]
```diff
@@ -390,3 +390,46 @@ fn test_sandbox_only_function() {
...
+    #[cfg(feature = "nightly")]
+    tb.expect(expect![[r#"
+        VMOutcome: balance 4 storage_usage 12 return data None burnt gas 48450963 used gas 48450963
+        Err: PrepareError: Error happened while deserializing the module.
+    "#]]);
+    #[cfg(not(feature = "nightly"))]
+    tb.expect(expect![[r#"
+        VMOutcome: balance 4 storage_usage 12 return data None burnt gas 0 used gas 0
+        Err: PrepareError: Error happened while deserializing the module.
+    "#]]);
```
[the remark this answers is not in the document]
Comment (nagisa) [reply runtime/near-vm-runner/src/tests/compile_errors.rs:413]: I sort of avoided looking too deeply into what exact reason for the difference is – when this change gets promoted to a proper feature (or is removed) the test will fail anyhow and the other branch then can be removed & I don't think it is beneficial to hold tests to the same level of rigour as we do for the regular kind of code in this instance.

[the remark this answers is not in the document]
Comment (jakmeier) [reply runtime/near-vm-runner/src/tests/compile_errors.rs:413]: ok fair enough, I guess we can merge it then

**Review by jakmeier — APPROVED** [review]

**Review by akashin — APPROVED** [review]
Can't provide much insight with my review, but the general premise makes sense to me

**Pull request merged**